src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,29 @@
 import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { filterSlice } from '../../features/filter';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const setStatus = (value: string) =>
+    dispatch(filterSlice.actions.setStatus(value));
+  const setQuery = (value: string) =>
+    dispatch(filterSlice.actions.setQuery(value));
+  const clearQuery = () => dispatch(filterSlice.actions.clearQuery(null));
+  const { query } = useAppSelector(state => state.filter);
+
   return (
     <form
       className="field has-addons"
       onSubmit={event => event.preventDefault()}
     >
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
+          <select
+            data-cy="statusSelect"
+            onChange={event => {
+              setStatus(event.target.value);
+            }}
+          >
             <option value="all">All</option>
             <option value="active">Active</option>
             <option value="completed">Completed</option>
```

Comment:
Can it be Enum and then we can use `Object.values(your enum` and `map()` methods